pkg/skaffold/docker/default_repo.go

```diff
@@ -19,13 +19,15 @@ package docker
 import (
 	"regexp"
 	"strings"
+
+	"github.com/docker/distribution/reference"
 )
 
 const maxLength = 255
 
 var (
 	escapeRegex = regexp.MustCompile(`[/._:@]`)
-	prefixRegex = regexp.MustCompile(`(.*\.)?gcr.io/[a-zA-Z0-9-_]+/?`)
+	prefixRegex = regexp.MustCompile(`^` + reference.DomainRegexp.String() + `/[a-z0-9]+/?`)
```

Comment:
This variable isn't clearly named, but it is used to match the GCR registry and project ID.  [console.cloud.google.com/projectcreate](https://console.cloud.google.com/projectcreate) further says:
> Project ID must be between 6 and 30 characters.
> Project ID can have lowercase letters, digits, or hyphens. It must start with a lowercase letter and end with a letter or number.

How about the following change?
```suggestion
         // gcpProjectIDRegex matches a GCP Project ID as according to console.cloud.google.com.
         gcpProjectIDRegex = `[a-z][a-z0-9-]{4,28}[a-z0-9]`
         // prefixRegex is used to match a GCR or AR reference, which must have a project ID.
	prefixRegex = regexp.MustCompile(`^` + reference.DomainRegexp.String() + `/` + gcpProjectIDRegex + `/?`)
```